### PR TITLE
updated apache2 template

### PR DESCRIPTION
**Pull request by allan-simon**
to support
- http authorization headers   (useful for http basic auth and oauth2 bearer token)
- being able to access to the production path  (mywebsite.com/backend/channel  without the app.php ) 
- living without the .htaccess (with provide slightly better performance)

### CODE REVIEW

**Comment by allan-simon** [comment]
@cbrunnkvist  does it looks good to you ?

File: provisioning/roles/webapp/templates/apache2.vhost.j2 [hunk 1]
```diff
@@ -1,14 +1,45 @@
 <VirtualHost *:80>
-    ServerName      Symfony
-    DocumentRoot    "{{ APPWEBROOT }}"
+    ServerName Symfony
+    ServerAlias Symfony
+
+    ErrorLog /var/log/apache2/project_error.log
```
Comment (cbrunnkvist) [inline provisioning/roles/webapp/templates/apache2.vhost.j2:5]: If you're gonna deviate from the main log, why not use APPNAME so it makes sense if you prov >1 app

File: provisioning/roles/webapp/templates/apache2.vhost.j2 [hunk 2]
```diff
@@ -1,14 +1,45 @@
 <VirtualHost *:80>
-    ServerName      Symfony
-    DocumentRoot    "{{ APPWEBROOT }}"
+    ServerName Symfony
+    ServerAlias Symfony
```
Comment (cbrunnkvist) [inline provisioning/roles/webapp/templates/apache2.vhost.j2:3]: (same as below, APPNAME) and why the Alias?

File: provisioning/roles/webapp/templates/apache2.vhost.j2 [hunk 3]
```diff
@@ -1,14 +1,45 @@
...
+    # used for http basic auth and oauth2 bearer token
+    SetEnvIfNoCase ^Authorization$ "(.+)" HTTP_AUTHORIZATION=$1
+    
     SetEnv          SYMFONY__DATABASE__USER {{ DBUSER }}
     SetEnv          SYMFONY__DATABASE__PASSWORD {{ DBPASSWORD }}
 
-    ProxyPassMatch ^/(.+\.(hh|php)(/.*)?)$ fcgi://127.0.0.1:9000/{{ APPWEBROOT }}/$1
+    DocumentRoot {{ APPWEBROOT }}
+    <Directory {{ APPWEBROOT }}>
+    AllowOverride None
```
Comment (cbrunnkvist) [inline provisioning/roles/webapp/templates/apache2.vhost.j2]: Indent this file consistently if you're gonna make it long

**Comment by cbrunnkvist** [comment]
If you are completely gonna override the `.htaccess`, then also make sure that existing/newly created applications don't ship with one

**Pull request merged**